chain/client/src/client.rs

```diff
@@ -850,66 +850,61 @@ impl Client {
         self.process_partial_encoded_chunk(MaybeValidated::from_validated(partial_chunk))
     }
 
+    /// Try to process chunks in the chunk cache whose previous block hash is `prev_block_hash` and
+    /// who are not marked as complete yet
+    pub fn check_incomplete_chunks(&mut self, prev_block_hash: &CryptoHash) -> Vec<AcceptedBlock> {
+        let mut accepted_blocks = vec![];
+        for chunk_header in self.shards_mgr.get_incomplete_chunks(prev_block_hash) {
+            debug!(target:"client", "try to process incomplete chunks {:?}, prev_block: {:?}", chunk_header.chunk_hash(), prev_block_hash);
+            // create a fake partial encoded chunk message with empty parts and receipts
+            // because we just want to reconstruct the chunk
+            let res = self.process_partial_encoded_chunk(MaybeValidated::from_validated(
+                PartialEncodedChunk::new(chunk_header, vec![], vec![]),
+            ));
```

Comment:
This piece of code feels suboptimal. Is there a way to do it without this hack?

Reply:
You were right. CI was failing because of the hack introduced a bug. I fixed it.